tpm2/tpm2_test.go

```diff
@@ -877,3 +877,57 @@ func TestEncodeDecodeCreationData(t *testing.T) {
 		t.Errorf("got decoded value:\n%v\nwant:\n%v", decoded, cd)
 	}
 }
+
+func TestCreateAndCertifyCreation(t *testing.T) {
+	rw := openTPM(t)
+	defer rw.Close()
+	params := Public{
+		Type:       AlgRSA,
+		NameAlg:    AlgSHA256,
+		Attributes: FlagFixedTPM | FlagFixedParent | FlagSensitiveDataOrigin | FlagRestricted | FlagSign | FlagNoDA | FlagUserWithAuth,
+		AuthPolicy: nil,
+		RSAParameters: &RSAParams{
+			Sign: &SigScheme{
+				Alg:  AlgRSASSA,
+				Hash: AlgSHA256,
+			},
+			KeyBits: 2048,
+			Modulus: big.NewInt(0),
+		},
+	}
+	keyHandle, pub, _, creationHash, tix, _, err := CreatePrimaryEx(rw, HandleEndorsement, pcrSelection, emptyPassword, emptyPassword, params)
+	if err != nil {
+		t.Fatalf("CreatePrimary 2 failed: %s", err)
+	}
+	defer FlushContext(rw, keyHandle)
+	attestation, signature, err := CertifyCreation(rw, emptyPassword, keyHandle, keyHandle, nil, creationHash, tix)
+	if err != nil {
+		t.Fatalf("CertifyCreation failed: %s", err)
+	}
+	att, err := DecodeAttestationData(attestation)
+	if err != nil {
+		t.Fatalf("DecodeAttestationData(%v) returned error: %v", attestation, err)
```

Comment:
keep error messages consistent: `Foo failed: %v`
replace all of: `returned error`, `returned err`, `Signature failed to verify`